travis-ymls/corosync.travis.yml

```diff
@@ -0,0 +1,69 @@
+# ----------------------------------------------------------------------------
+#
+# Package             : corosync
+# Source Repo         : https://github.com/corosync/corosync
+# Travis Job Link     : https://travis-ci.com/github/gururajrkatti/corosync/jobs/466439735
+# Created travis.yml  : Yes
+# Maintainer          : Gururaj R Katti <Gururaj.Katti@ibm.com>
+#
+# Script License      : Apache License, Version 2 or later
+#
+# ----------------------------------------------------------------------------
+
+language: c
+
+arch:
+   - amd64
+   - ppc64le   
+
+addons:
+    apt: 
+     update: true
+     packages:
+        - libnss3-dev 
+        - libtool
+        - libsctp-dev 
+        - check 
+        - pkg-config 
+        - groff 
+        - libpam0g-dev 
+        - liblz4-dev 
+        - doxygen 
+        - zlib1g 
+        - zlib1g-dev 
+        - libssl-dev 
+        - liblzo2-dev 
+        - liblzma-dev 
+        - libbz2-dev 
+        - libxml2-dev
+        - libzstd-dev
+        - libnl-3-dev
+        - libnl-route-3-dev
+
+before_script: 
+    # Install libqb
+    - cd /tmp
+    - git clone git://github.com/asalkeld/libqb.git && cd libqb
+    - ./autogen.sh
+    - ./configure
+    - make
+    - make check 
+    - sudo make install
+    # Install libknet.
+    - cd /tmp
+    - git clone https://github.com/fabbione/kronosnet && cd kronosnet
+    - ./autogen.sh
+    - ./configure
+    - sed -i -e 's/-Werror//' libknet/tests/Makefile
+    - sed -i -e 's/LIBS = /LIBS = -lpthread /' libknet/tests/Makefile
+    - make
+    - sudo make install
```

Comment:
Did you try using `libqb-dev` & `libknet-dev` from ubuntu repos instead of building those from source?. You might have to using bionic `distro` for this.

- https://packages.ubuntu.com/bionic/libqb-dev
- https://packages.ubuntu.com/bionic/libknet-dev

Reply:
I had referred the build script and written this. Have incorporated your suggestions.